### PR TITLE
Problem: Protobuf code generation is not consistent with cosmos-sdk

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Generate proto code using buf with similar configuration as cosmos-sdk

### CODE REVIEW

File: proto/buf.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This module represents the "proto" root.
+version: v1
+name: buf.build/crypto-org-chain/chain-main
```
Comment (tomtau) [inline proto/buf.yaml:3]: this name/url gives 404 -- do we need to register it on buf.build?

**Review by tomtau — APPROVED** [review]
it seems the third party Git sub-modules are no longer needed after this (the needed protobuf definitions are fetched via buf?)? so could they be removed?

File: proto/buf.yaml (hunk 1 above)
Comment (devashishdxt) [reply proto/buf.yaml:3]: https://docs.buf.build/bsr/usage#push-a-module

**Comment by devashishdxt** [comment]
> it seems the third party Git sub-modules are no longer needed after this (the needed protobuf definitions are fetched via buf?)? so could they be removed?

They're still needed for swagger documentation.

File: proto/buf.yaml (hunk 1 above)
Comment (tomtau) [reply proto/buf.yaml:3]: ok, for releases, we can perhaps specify an action to do it: https://docs.buf.build/ci-cd/github-actions

Comment (devashishdxt) [reply proto/buf.yaml:3]: Created #852

**Comment by yihuang** [comment]
Shall we update the `config.json` too? similar to https://github.com/crypto-org-chain/cronos/pull/677/files#diff-b8ef6a5786bfd5df3147645f82844be943e1fe6d68de3b6b4681a941e077b8d7

**Comment by devashishdxt** [comment]
> Shall we update the `config.json` too? similar to https://github.com/crypto-org-chain/cronos/pull/677/files#diff-b8ef6a5786bfd5df3147645f82844be943e1fe6d68de3b6b4681a941e077b8d7 The legacy APIs are removed.

Done

**Comment by yihuang** [comment]
> > Shall we update the `config.json` too? similar to https://github.com/crypto-org-chain/cronos/pull/677/files#diff-b8ef6a5786bfd5df3147645f82844be943e1fe6d68de3b6b4681a941e077b8d7 The legacy APIs are removed.
> 
> Done

you need to regenerate the swagger to make it up to date.

**Comment by devashishdxt** [comment]
> > > Shall we update the `config.json` too? similar to https://github.com/crypto-org-chain/cronos/pull/677/files#diff-b8ef6a5786bfd5df3147645f82844be943e1fe6d68de3b6b4681a941e077b8d7 The legacy APIs are removed.
> > 
> > 
> > Done
> 
> you need to regenerate the swagger to make it up to date.

done

**Comment by yihuang** [comment]
> > > > Shall we update the `config.json` too? similar to https://github.com/crypto-org-chain/cronos/pull/677/files#diff-b8ef6a5786bfd5df3147645f82844be943e1fe6d68de3b6b4681a941e077b8d7 The legacy APIs are removed.
> > > 
> > > 
> > > Done
> > 
> > 
> > you need to regenerate the swagger to make it up to date.
> 
> done

and statik needs to be updated too ;D

**Comment by devashishdxt** [comment]
> and statik needs to be updated too ;D

Done

**Review by yihuang — APPROVED** [review]
LGTM

**Pull request merged**